### PR TITLE
feat: supports concurrency config

### LINKED ISSUE #36: It would be nice to be able to set 'concurrency' when creating `batch` instance
Currently I do:

```
      const batch = new Batch();
      batch.concurrency(20);
```

I would like to have the ability to do:

```
      const batch = new Batch({ concurrency: 20 });
```

### PR DESCRIPTION
**Pull request by schne324**
resolves #36

### CODE REVIEW

**Comment by dougwilson** [comment]
Awesome, thanks 👍 Can you mention this new API in the Readme somewhere as well?

**Comment by schne324** [comment]
No problem! Just updated the readme, let me know if you think it is sufficiently documented

**Comment by schne324** [comment]
@dougwilson what do you think of the docs?

File: index.js [hunk 1]
```diff
@@ -33,10 +33,11 @@ module.exports = Batch;
  * Create a new Batch.
  */
 
-function Batch() {
+function Batch(config) {
   if (!(this instanceof Batch)) return new Batch;
```
Comment (andrzej-aa) [inline index.js]: It should be `return new Batch(config)` to not lose the configuration

File: index.js [hunk 2]
```diff
@@ -33,10 +33,11 @@ module.exports = Batch;
...
+  config = config || {};
   this.fns = [];
-  this.concurrency(Infinity);
+  this.concurrency(config.concurrency || Infinity);
   this.throws(true);
   for (var i = 0, len = arguments.length; i < len; ++i) {
     this.push(arguments[i]);
```
Comment (andrzej-aa) [inline index.js]: This is pushing the config as one of the tasks

**Review by dougwilson — APPROVED** [review]
sorry this got lost in time

**Pull request merged**